src/KafkaFlow/Consumers/MessageConsumer.cs

```diff
@@ -89,37 +84,37 @@ public async Task RestartAsync()
 
         private async Task InternalRestart()
         {
-            await this.consumer.StopAsync().ConfigureAwait(false);
+            await this.consumerManager.StopAsync().ConfigureAwait(false);
             await Task.Delay(5000).ConfigureAwait(false);
```

Comment:
Why this 5 s delay?

Reply:
This time was used before the refactoring. It is there to wait for a time for Kafka complete the partitions rebalance process before starting again.

Reply:
Right, however, if the rebalance delays more than 5 sec (what is not usual) then maybe the system could stay in an unpredictable state.
I know it is not related to this MR but, how this is a big refactoring, we could think about using the SetPartitionsAssignedHandler on ConsumeBuilder to remove this fixed time delay.
What do you think?

Reply:
PS: the same on Consumer.ConsumeAsync(CancellationToken cancellationToken)

Reply:
If the stop takes more than 5s, another rebalance will be executed, which is not a problem exactly. It is more to avoid too many rebalances.